docs/current_state_specification.md

```diff
@@ -0,0 +1,153 @@
+
+### Current State Data Model
+
+The following table describes all supported properties and sections of the Current State specification:
+
+| Property | Type | Description |
+| - | - | - |
+| **General properties** | | |
+| hardwareNodes | JSON array | Inventory for a list of hardware nodes |
+| softwareNodes | JSON array | Inventory for a list of software nodes |
+| associations | JSON array | List of mappings between the inventory nodes |
+| **Hardware node properties** | | |
+| id | string | Identifier of the hardware node |
+| version | string | Version of the hardware node |
+| name | string | Name of a hardware node |
+| parameters | JSON array | List of key/value parameters for a hardware node |
+| addressable | boolean | Enables hardware node addressability |
+| **Software node properties** | | |
+| id | string | Identifier of the software node |
+| version | string | Version of the software node |
+| name | string | Name of the software node |
+| parameters | JSON array | List of key/value parameters for a software node |
+| type | string | Type of the software node. The supported types are listed below |
+| **Parameter properties** | | |
+| key | string | Key of the parameter |
+| value | string | Value of the parameter |
+| **Association parameter** | | |
+| sourceId | string | Identifier of the source node of the association |
+| targetId | string | Identifier of the target node of the association |
+
+### Supported software types
+
+The list of the supported software types :
+
+| Type | Description |
+| - | - |
+| IMAGE | Represents an image software type |
+| RAW | Represents a raw bytes software type |
+| DATA | Represents a data software type |
+| APPLICATION | Represents an application software type |
+| CONTAINER | Represents a container software type |
+
+### Device Inventory Graph Representation
+
+The diagram below represents the Device Inventory graph and the links between the software and hardware nodes.
+
+The software nodes are organized in the graph at different levels. At the root level stands the main Update Manager software node, which represents the Update Manager component. At the second level are placed the root nodes for each supported domain agent, which are linked with associations to the Update Manager node. At the last level in the hierarchy are placed the domain-specific software nodes, representing the domain components. These software nodes can be modeled the in a tree-based structure if the internal domain-specific representation is more complex.
+
+The hardware nodes do not follow any strict hierarchy and can be linked to any hardware or software node. Any cycles between the nodes are not allowed and prevented.
+
+![Device inventory](/_assets/device-inventory.svg)

```

Comment:
The link seems to be broken. My guess is that should be either `/docs/_assets/device_inventory` or `./_assets/device_inventory` .